wandb/old/settings.py

```diff
@@ -106,9 +107,12 @@ def _settings(default_settings={}):
 
     @staticmethod
     def _global_path():
-        config_dir = os.environ.get(
-            env.CONFIG_DIR, os.path.join(os.path.expanduser("~"), ".config", "wandb")
-        )
+        default_config_dir = os.path.join(os.path.expanduser("~"), ".config", "wandb")
+        # if not writable, fall back to a temp directory
+        if not os.access(default_config_dir, os.W_OK):
+            default_config_dir = os.path.join(tempfile.gettempdir(), ".config", "wandb")
```

Comment:
This means that settings will *not* be persisted. This is dangerous, we should at least print a warning.